EIPS/eip-2045.md

```diff
@@ -0,0 +1,119 @@
+
+## Motivation
+The transaction capacity of an Ethereum block is determined by the gas cost of transactions relative to the block gas limit. One way to boost the transaction capacity is to raise the block gas limit. Unfortunately, raising the block gas limit would also increase the rate of state growth, unless the costs of state-expanding storage opcodes (SSTORE, CREATE, etc.) are simultaneously increased to the same proportion. Increasing the cost of storage opcodes may have adverse side effects, such as shifting the economic assumptions around gas fees of deployed contracts, or possibly breaking invariants in current contract executions (as mentioned in EIP-2035 [REF/LINK HERE], more research is needed on the potential effects of increasing the cost of storage opcodes).
+
+Another way to boost the transaction capacity of a block is to reduce the gas cost of transactions. Reducing the gas costs of computational opcodes while keeping the cost of storage opcodes the same, is effectively equivalent to raising the block gas limit and simultaneously increasing the cost of storage opcodes. However, reducing the cost of computational opcodes might avoid the adverse side effects of an increase in cost of storage opcodes (again, more research is needed on this topic).
+
+Currently, computational opcode costs are already too close to the minimum unit of 1 gas to achieve the large degree of cost reductions that recent benchmarks<sup>[1](#evmbenchmarks)</sup> indicate would be needed to tune opcode gas costs to the performance of optimized EVM implementations. A smaller minimum unit called a "particle"<sup>[2](#particle)</sup>, which is a fraction of 1 gas, would enable large cost reductions.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+A new gas counter `particlesUsed` is added to the EVM, in addition to the existing gas counter `gasUsed`. The unit 1 gas is equal to 10000 particles (`PARTICLES_PER_GAS`). The `particlesUsed` counter is only increased for opcodes priced in particles (i.e. opcodes that cost less than 1 gas). If `particlesUsed` exceeds 1 gas, then 1 gas is added to `gasUsed` (and deducted from `particlesUsed`).
```

Comment:
I'd argue to follow the Ewasm proposal and only account in "particles". Instructions which aren't reduced to particles levels are just scaled up, for example `SSTORE` would cost `5000000` assuming `PARTICLES_PER_GAS = 10000`.

At the end of execution scale back particles to gas: `ceil(particlesUsed / PARTICLES_PER_GAS)`.

Reply:
I tried to describe this in the next paragraph: "A more performant implementation might instead keep a single `gasUsed` counter, multiply opcode costs by 10000 and the `gasLimit` by 10000, ...".  Will push an edit to clarify.

Reply:
Realised that after writing and pushed a clarification myself to https: https://github.com/axic/EIPs/tree/gas-particles/EIPS